### DISCUSSION TITLE
why does my 2 pixel size outline looks blurry

### DISCUSSION THREAD

**Discussion by HSDSZ** [bnpr/Malt#552] [Q&A]
Here is the outline width = 1 pixel, which looks normal
![image](https://github.com/bnpr/Malt/assets/36077492/a4bf7886-f3db-428f-9130-e60cbbf3eeea)

If I change the width to 2, I suppose i could get a sharp 2 pixel outline, but the result is kind of blurry
![image](https://github.com/bnpr/Malt/assets/36077492/8565d1a5-8da5-4f99-8fed-61ed40ff78fd)

here is the render setting.
![image](https://github.com/bnpr/Malt/assets/36077492/65f6c168-60cc-4a13-a88b-75afd34861af)

**Comment by pragma37** [comment]
Technically, the line is centered at one pixel, so a 2 pixel wide line will cover 100% of that pixel and 50% of the contiguous ones (so 1.0 + 0.5 + 0.5 = 2.0).

I know it can seem strange, but I've made tests in the past where I've tried to "push" even-width lines to one side so they're aligned to the pixel grid and the transitions between different line widths look really bad.
